### DISCUSSION TITLE
Bringing back some old mods.

### DISCUSSION THREAD

**Discussion by PeaceIsBliss** [EnigmaticaModpacks/Enigmatica9#156] [Ideas]
Now I do understand some mods like Botania and Tinkers are just not ready; however, Dark Utils, Integrated Dynamics, Pedestals, and Waystones have all been updated and still missing from the pack despite being released prior to 1.3 deployment. Can we expect these classics to come back to the pack?

**Comment by MuteTiefling** [comment]
We're attempting to keep the mod list lighter in e9. We invite you to expand your horizons by finding new ways of doing the things you've done before. 

Dark Utilities: Aside from the sleep charm, everything is replicated by other mods. As far as the sleep charm goes, there are wind chimes that will prevent Phantoms from attacking you while you're near the chime. Putting a few around your base will keep you equally safe while not removing the danger in the wild.

Integrated Dynamics: The automation and storage aspects of this are amply covered by AE2 as well as other larger content mods suchs as PNC, IE, Ars Nouveau, and others. The machines are all covered by other mods as well. 

Pedestals: This one is still up for discussion, but it's going to depend on a lot of factors and will be determined internally if it is included or not.

Waystones: This mod is entirely duplicated by things like Ars Nouveau and Mekanism. 

In short, they are not missing and were not forgotten. We've opted out of including them.

**Comment by PeaceIsBliss** [comment]
Obviously I went into more detail on Discord, but I am saddened by this. A shift in direction for the pack is entirely your right, but does not appeal to me in its entirety. I as of yet have not felt much support for my standpoint from the Enigmatica community so as a minority my opinion holds little weight. Thank you for the response.

**Reply by PeaceIsBliss** [reply]
And I strongly suggest reconsidering Waystones. Not all servers are equal, and for instance on ours, with one /home to start, its been a landgrab for meteorites with perfect crystals and stronghold end portals, since noone can provide a public location to teleport to unless you /tpa a player. We get one FTB loaded chunk (which is perfectly fine as players build the TPS goes way down without limits). But it throws MEK portals right out the window. And I'm pretty sure Ars doesnt go cross-dimensional. Plus sharestones are just nice to have if your server people like making public farms. Would be seriously restricting the playability on many servers without something like this. Or force the server to add another server-side command mod themselves. - just a thought.

**Reply by NielsPilgaard** [reply]
You make a compelling case 😄 What do you think @MuteTiefling? Bring back Waystones, but up xp cost perhaps?

**Comment by MuteTiefling** [comment]
Ok, just to try to bring this all back together since you've now posted two separate requests here on Github as well as continuing one in Discord... I'd also prefer to keep this here where we can reference it easily the next time it comes up.

>Dont miss Ultimine at all honestly.
But I'm back!
So reflecting on our last conversation since I was extremely tired and not entirely sober. I still feel my initial analysis stands and has been unanswered. Why so many item restrictions?
all machines from Engineers Decor been disabled, most of RFTools (but left quarry cards and builder??)
This whole expand my horizons thing is sillyness. As I use practically every mod in some way
also, Waystones is very important for servers that restrict chunk loading. as MEK becomes useless at that point
I use InDy to auto filter drops from mobs farms for instance by mod embeds
not for storage
I use RFTools Sensor to detect the lack of wither so the other mods dont duplicate drops attempting to create one before space is available.
its all about utility that other mods dont provide. essentially
and I will say this for absolute certainty; the huge appeal of Enigmatica to the groups ive played with in past, several severs, different versions, family and friends: is the fact that any style player can play and enjoy playing. Magic, Machinist. Explorer. Farmer. Whatever, everyone enjoys it. Now it feels like that wont be the case anymore
So that my harumph. Sorry about poorly expressing it last time. And I want to make clearly i absolutely enjoy the other packs and your hard work.
and on a side note, i never even knew about Spirit, and totally going to check that out for spawners. So yeah cardboard box restriction makes more sense now. Though I dont think it works on all chests since I have heard complaints from people on our server. But I have not tested that thoroughly enough to have input

That's a lot to unpack, so let me go step by step. 

> all machines from Engineers Decor been disabled

Engineer's Decor is a really neat mod. I love it, honestly, but it has one glaring flaw: Discoverability. None of its machines show up in JEI with what they do. Searching up how to make Magma Blocks, for example, should bring up the Small Mineral Melter. Instead, you'd only know what it does by digging through the book or the tooltips.

This is the reason it was rejected initially. 

It was pointed out that it still has a lot of unique decor items, so it was reconsidered and brought in for that reason only. Since the machines are not easily discoverable, we opted out of including them.

> most of RFTools

Niller already covered this one amply. 

>  Waystones is very important for servers that restrict chunk loading. as MEK becomes useless at that point

Just as we don't support issues that result from a modification to the pack, I don't feel that we should be obliged to include mods due to choices made by server operators. Chunk loading is a fundamental concept in Minecraft and like it or not, it's a requirement for things to work properly. If the server operators have decided to alter the pack by disabling chunk loading or severely limiting it, then finding solutions to the problem they've created is on them. They could, for instance, disable the recipe for Anchor Upgrades but leave Mekanism's chunk loading enabled, thus allowing an OP to step in and set up a communal portal. 

Similarly, without the need of any OP, the players could set up a portal network with Ars Nouveau that would require no chunk loading. Yes, you may have to take a few portals to get to your ultimate destination if it's in another dimension, but it's still possible to traverse dimensions with standard portals. And Ars Portals can even dump you directly inside those portals so one step takes you to the new dimension.

Lastly, while not teleportation, Create has quite possibly one of the coolest long distance transportation methods and the creative possibilities of Trains are endless. And yes, they can pass through portals. 

**Honestly, I'd just as soon keep Waystones out of the pack as it's one of those things that immediately obsoletes all other options in a way that doesn't promote player creativity.**

> Not all servers are equal, and for instance on ours, with one /home to start, its been a landgrab for meteorites with perfect crystals and stronghold end portals, 

There are 128 Strongholds in every world and countless meteors. I think that you guys can share a bit. One small communal chunkloaded Certus farm would probably supply you all indefinitely. It's also entirely possible to automate crafting the Flawed Budding Crystals, placing them, and breaking them when they revert. So a perfect crystal is not a requirement by any means. You just need a little bit to get you going.

I would not be opposed to adding recipe solutions to help with the Certus issue, however, and it's been something I've had on my mind for quite some time. I'm simply trying to gauge how necessary it actually is before I pull the trigger on that one. Your feedback in this regard is helpful. There's already one change to it implemented in Dev to help somewhat, making use of the Thermal Crystallizer. 

Adding Waystones is not the only solution to this problem; it's not even the best solution. AE2 lets you move those crystals by way of Spatial Storage, so you can always go that route. I wouldn't be terribly surprised if a Create Contraption could move one without harming it too, for that matter. 

> I use InDy to auto filter drops from mobs farms for instance by mod embeds not for storage

The quests go to great lengths to detail all of your logistical options from each mod. You'll see a fair bit more of this coming next update. Integrated Dynamics is not the only mod capable of sorting items. You've got options here and I invite you to explore something new. I know from reading your comments in Discord that this one is more to do with the 20 hours you spent writing a script that you now feel is wasted, however, as well as a misunderstanding of how Apotheosis works in 1.19 which obviates the need to farm specific affixes to begin with.

If none of the more traditional sorting options fit your needs, there's always ComputerCraft too.

> I use RFTools Sensor to detect the lack of wither so the other mods dont duplicate drops attempting to create one before space is available. its all about utility that other mods dont provide. essentially

Industrial Foregoing has a simple Mob Detector that replicates this exact functionality. PneumaticCraft also has a Universal Sensor which requires a little bit of effort to set up and run, but that works wonderfully. 

One could even rig together something far more interesting by using Ars Nouveau Turrets. `Projectile > Redstone Signal` shot through the area where a Wither is supposed to be. It would hit the Wither and stop there. If there's no Wither, it could pass unimpeded and strike some Redstone on the other end, enabling it and triggering building another Wither.

> This whole expand my horizons thing is sillyness. As I use practically every mod in some way 

If E6E taught me anything it's that people are resistant to change, but are often thankful for being forced to look for alternatives. I've seen this sentiment come up regularly on Discord: "I've always avoided mod X for Y reasons, but having to go through it for Expert showed how interesting it can be". 

I'd also like to take a moment to directly address some of the other issues you brought up on Discord.

No trash can. 

- Thermal Nullifier for items and fluids. 
- Mekanism has a void mode on chemical tanks for gases. 
- A cauldron with lava can also void items, and [Supplementaries](https://github.com/MehVahdJukaar/Supplementaries/wiki/Redstone-Blocks#faucet) has some neat interactions as well. 
- Create Crushers or Immersive Engineering Crushers could even be a fun alternative as they destroy anything that doesn't have a recipe...

RFtools Shield as the only player damage option.

- Industrial Foregoing Mob Crusher
- Ars Nouveau Turrets
- Pneumaticcraft Drones
- Create Deployers
- Ars Nouveau Drygmies that don't need to kill to get player drops

Feeling forced into all magic.

- There are, quite simply, more Tech mods in the pack than Magic at this point. Being a kitchen sink pack, there's also no forcing you to do anything. 
![image](https://user-images.githubusercontent.com/9543430/205523552-9a577f1a-a180-41ce-8797-d1c270e6f5cc.png)
I admit that some things will currently only have a few solutions, like Certus quartz, but as mod integration efforts continue, more mods will be able to do more things, giving you far more freedom in how you approach a task. That takes time, but is well under way. 

Spirit as the only mob spawner.

- PneumaticCraft Pressurized Spawner
- Industrial Foregoing Mob Duplicator
- Ars Nouveau has Drygmies that obsolete the need of spawning mobs in the first place. You an even put the mobs in jars, completely removing the entities from the world.
- Gateways to Eternity have been customized (and continue to be customized) to add interesting gateways that net you an enormous amount of drops. Look them up. Yes, they need Occultism, but they already cover a lot of the mobs you might want to spawn.

Lack of Ranged Pickup options.

- Ars Nouveau Spell Turrets (seriously, if you like Integrated Dynamics as much as it seems you do, you should give these things a try)
- Nature's Aura Hopper Enhancement
- Thermal Vacuumulators 
- PneumaticCraft Collection Drones or programmable drones
- Sophisticated Storage and Backpacks Pickup upgrades
- Ars Nouveau Starbuncles
- Create Ploughs could be used to sweep items into a collection area as well
- Mob Crushers automatically pick up the drops from their kills as well

Being forced into villagers and villagers being a pain due to Apotheosis.

- Villagers have had their enchantment selection tremendously pared down specifically to combat both the feeling that villagers are required and that they take forever to get anything good due to the vast number of enchants. Put simply, they don't have access to sell every enchant. An automated villager re-roller should be a thing of the past as they just don't have that many options to pick from.
- You'll find that the far better source of enchants is actually from mobs as a number of them been given custom gear to both toughen them up and reward people with the enchanted loot. This is probably broken n the current release, but it's fixed for next release. Prepare to encounter a bunch of custom equipped mobs soon. 
- Industrial Foregoing and PneumaticCraft both also have some really nice tools for working with Enchants, in addition to those offered by Apotheosis. 

Lack of Armor Choices.

- PneumaticCraft Armor - highly customizable before getting into Apotheosis affixes
- Ars Nouveau Armor - highly customizable before getting into Apotheosis affixes
- Mekanism MekaSuit - highly customizable before getting into Apotheosis affixes
- Many other options, just search `$armor` in JEI, they've all been tagged for easy discovery

You state over and over that you're dissappointed by the lack of options in the pack, yet you keep asking for the same things that you're familiar with instead of actually exploring the options that are present. 

I understand that change is hard. Breaking out of your comfort zone is always difficult. That is a notion that I'm keeping in the back of my mind as I continue to update quests and categorize items via tags and custom tooltips. I fully understand that players won't be familliar with all of these options, but to state that they're not present is simply false. You'd be better served by maintaining an open mind and asking how to achieve a certain goal than to go in assuming something is impossible because you don't see an immediate known solution. The Discord is full of very helpful and knowledgeable players. 

For my part, I'll take this feedback in consideration as I move foreward with integration work. I see a number of places where tags or tooltips would help with discoverability, for instance, and things that may need more love in quests. 

As I explained the first time we spoke, the pack is a work in progress and things are still very early. If you had looked at e6 this early in its development, you would have likely had a similar reaction. I still remember when my only option for safely encasing a Wither to kill it was to use Modular Routers as a cage since they have an upgrade to be immune to the Wither. Give it some time. There are still mods that we want to add that haven't even ported yet.
